### ISSUE TITLE
Unable to retrieve child resources.

### ISSUE THREAD

**Issue by fratiigrimm** [microsoft/AzureStorageExplorer#6074]
Unable to retrieve child resources.

Details:
{
  "name": "RestError",
  "message": "getaddrinfo ENOTFOUND sastaticypii2ymfqcrzy.table.core.windows.net",
  "stack": "RestError: getaddrinfo ENOTFOUND sastaticypii2ymfqcrzy.table.core.windows.net\n    at ClientRequest.<anonymous> (C:\\Users\\frati\\AppData\\Local\\Programs\\Microsoft Azure Storage Explorer\\resources\\app\\node_modules\\se-table-extension\\dist\\src\\index.js:2:180083)\n    at Object.onceWrapper (node:events:646:26)\n    at ClientRequest.emit (node:events:526:28)\n    at TLSSocket.socketErrorListener (node:_http_client:442:9)\n    at TLSSocket.emit (node:events:526:28)\n    at emitErrorNT (node:internal/streams/destroy:157:8)\n    at emitErrorCloseNT (node:internal/streams/destroy:122:3)\n    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)",
  "code": "ENOTFOUND",
  "request": {
    "url": "https://sastaticypii2ymfqcrzy.table.core.windows.net/Tables",
    "headers": {
      "accept": "application/json;odata=minimalmetadata",
      "x-ms-version": "2019-07-07",
      "dataserviceversion": "3.0",
      "accept-encoding": "gzip,deflate",
      "user-agent": "Microsoft Azure Storage Explorer/1.25.1 (win32) azsdk-js-data-tables/13.1.2 core-rest-pipeline/1.9.0 Node/v16.14.2 OS/(ia32-Windows_NT-10.0.22000)",
      "x-ms-client-request-id": "c36587df-6835-4f93-b021-3e8c41ae2f3c",
      "x-ms-date": "Mon, 05 Sep 2022 09:41:53 GMT",
      "authorization": "SharedKeyLite sastaticypii2ymfqcrzy:HT49wtYLVBOGjnu5AN74PrTnMySwnB1q09OS6wSXIak="
    },
    "method": "GET",
    "timeout": 0,
    "disableKeepAlive": false,
    "streamResponseStatusCodes": {},
    "withCredentials": false,
    "tracingOptions": {
      "tracingContext": {
        "_contextMap": {}
      }
    },
    "requestId": "c36587df-6835-4f93-b021-3e8c41ae2f3c",
    "allowInsecureConnection": true,
    "enableBrowserStreams": false
  }
}

**Comment by lizzbazan** [comment]
Same issue, please solve this, this started after I installed the last version 1.25.1

**Comment by ilyaploshchik** [comment]
Same issue persists! I hvae 1.25.1 installed as well.

**Comment by LightDiode** [comment]
The same issue. Cannot get any blobs from the emulated storage (the only way to get them - by the name)

**Comment by oskargonz** [comment]
I have the same without installing 1.25.1

**Comment by vidamaghami** [comment]
Same issue persists! I have 1.25.1 installed as well.

**Comment by MRayermannMSFT** [comment]
@fratiigrimm does not appear to be having an issue with their emulator.

@lizzbazan @ilyaploshchik @LightDiode @oskargonz @vidamaghami if you are having emulator issues and are using the old .NET emulator (aka, not Azurite) that is a known issue which you can learn more about here: https://github.com/microsoft/AzureStorageExplorer/issues/6030

Can each of you confirm if you are working with an emulator or not? If you are then please monitor the linked issue. If you are not, then please try what I'm about to suggest.

@fratiigrimm and anyone not working with an emulator, please try enabling system proxy: https://docs.microsoft.com/en-us/azure/storage/common/storage-explorer-network#use-system-proxy-preview

**Comment by co-dax** [comment]
Hey guys, I just had this issue today and found out that it worked after I uninstalled Azure Storage Emulator and and used Azurite instead of it (instead of Azure Storage Emulator).
It seems that latest version of Azure Storage Explorer is no compatible with Azure Storage Emulator but on only with Azurite. At the [this](https://docs.microsoft.com/en-us/azure/storage/common/storage-use-azurite?tabs=visual-studio) link it says:
 

> Azurite is the future storage emulator platform. Azurite supersedes the [Azure Storage Emulator](https://docs.microsoft.com/en-us/azure/storage/common/storage-use-emulator). Azurite will continue to be updated to support the latest versions of Azure Storage APIs.

...so perhaps the latest version of Azure Storage Emulator is meant to be working with Azurite. 

This is just a guess and it worked for me today.

**Comment by jinglouMSFT** [comment]
@co-dax Just FYI, Azure Storage Emulator has been deprecated if you are not aware of.  See the official doc here: https://docs.microsoft.com/en-us/azure/storage/common/storage-use-emulator

**Comment by co-dax** [comment]
@jinglouMSFT Yeah that's what I just discovered as mentioned in my last comment above. The point is that people, just like me, could still be using Azure Storage Emulator in combination with Azure Storage Emulator 1.25.1 which turned out to be creating the error from this thread for me. I had no more this error with Azure Storage Emulator 1.25.1 once I switched to Azurite.

**Comment by jinglouMSFT** [comment]
@co-dax The error you were experiencing with the Azure Storage Emulator is likely caused by the $blobchangefeed container, which is new and is not supported by the deprecated Storage Emulator.  We are thinking about filtering out this container when emulators are used in our next major release to unblock those customers who can't move from the Azure Storage Emulator to Azurite.  However, at some point down the road that more and more features will be added in the Storage services that the deprecated emulator would not work with.  As a long-term solution, we strongly recommend migrating to Azurite.

**Comment by co-dax** [comment]
@jinglouMSFT great, thanks.

**Comment by WillGoldLO** [comment]
28-SEP-2022


Unable to retrieve child resources.

Details:
{
  "name": "FetchError",
  "message": "request to http://127.0.0.1:10001/devstoreaccount1?comp=list&include=metadata&timeout=30 failed, reason: connect ECONNREFUSED 127.0.0.1:10001",
  "stack": "FetchError: request to http://127.0.0.1:10001/devstoreaccount1?comp=list&include=metadata&timeout=30 failed, reason: connect ECONNREFUSED 127.0.0.1:10001\n    at ClientRequest.<anonymous> (C:\\Users\\wdavis\\AppData\\Local\\Programs\\Microsoft Azure Storage Explorer\\resources\\app\\node_modules\\node-fetch\\lib\\index.js:1491:11)\n    at ClientRequest.emit (node:events:526:28)\n    at Socket.socketErrorListener (node:_http_client:442:9)\n    at Socket.emit (node:events:526:28)\n    at emitErrorNT (node:internal/streams/destroy:157:8)\n    at emitErrorCloseNT (node:internal/streams/destroy:122:3)\n    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)",
  "type": "system",
  "errno": "ECONNREFUSED",
  "code": "ECONNREFUSED"
}

**Comment by MRayermannMSFT** [comment]
@WillGoldLO it looks like you're trying to work with an emulator. Please make sure you have actually started the emulator. Storage Explorer will not start it for you.

**Comment by MRayermannMSFT** [comment]
I'm going to close this issue as the original author has not replied recently. Several troubleshooting suggestions have been commented. If you are continuing to face problems after trying them please open your own issue. Thanks!

**Comment by WillGoldLO** [comment]
So you make a comment on this issue after 3 days and close it as "author
has not responded" on the same day.  Seems like a lot of effort has gone
into troubleshooting.  Some of us have day jobs where we are using tools
that don't quite function as expected so we can't stop what we are doing to
respond immediately.

The emulator I am using is azurite. Running locally. And this problem has
not been resolved. But close the ticket by all means and let someone else
open another with the same unresolved issue.  Until this is fixed there
will continue to be issues reported.  There are right ways to take
advantage of the benefits of feedback from the community and there are ways
to brush the issues under the rug.

**Comment by jinglouMSFT** [comment]
@WillGoldLO your issue looked different from the original author of this issue, and since we haven't heard any reply from the original author of this issue for more than 28 days, our triage committee is ok to close the issue.  If MRayermannMSFT's suggestion or my earlier suggestion didn't work for you, please feel free to open a dedicated issue (to what you've experienced) so we can follow up and try to help.   While opening the issue, could you also please let us know which version of Storage Explorer you are using, Azurite version, operating system, and your connection string used to attach to the emulator (please remove any PII infor before sharing). Also share other info that might be able to help diagnose the issue.  This likely could help speed up the diagnose process.  Thank you and we appreciate your patience.

**Comment by MRayermannMSFT** [comment]
Hi @WillGoldLO to reiterate what jinglouMSFT has said, I closed the issue because of:

1. Lack of responses from people being affected by the issue indicated by the original issue author (we typically try to give people about a month before we go ahead and close issues)
2. The fact that the issue indicated by the original issue author is also a duplicate of an earlier issue (https://github.com/microsoft/AzureStorageExplorer/issues/6030)

We try to keep issues focused to one issue at a time to avoid confusion. We would love for you to open your own issue, one specific to getting an "ECONNREFUSED" error when trying to work with Azurite. Like jinglouMSFT said, please include the version of Storage Explorer you are using, Azurite version, operating system, and your connection string used to attach to the emulator (please remove any PII infor before sharing).

Sorry for the miscommunication!

**Comment by WillGoldLO** [comment]
This issue is duplicated in more than one thread and it is not resolved but these keep getting closed.  I commented on the duplicate 6030 which is also closed.  If I have to open my own even though it is a duplicate I will.  this is an aggravating new situation (last 3 months) that I want to see resolved.  I will help with info any way I can.

https://github.com/microsoft/AzureStorageExplorer/issues/6030#issuecomment-1271716795
